### PR TITLE
Karla rubiano refactor MVP

### PR DESCRIPTION
**Pull request by kdrubianoc**
#### What does this PR do?
Feat: Refactor MVVM to MVP in the ViewController. 
Update Unit Test according to the changes.

#### Where should the reviewer start?
MovieListPresenter, ViewController, MockMovieListView, MovieListPresenterTest.

#### Code review checklist [Code Review Rubric](https://github.com/wizeline/iOS-apprenticeship-program-2020/blob/main/code-review-rubric.md)

1. DRY Principle (Don’t repeat yourself)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

2. Yagni Principle (You Aren't Gonna Need It)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

3. KISS Principle (Keep It Simple, St...d)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

4. Remove unused code

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

5. Hard-coded values

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

6. Language style guide & coding conventions
 
 - [Wizeline Style Guide](https://github.com/wizeline/mobile-documentation/blob/master/guidelines/swift-style-guide.md) 
 - [RayWenderlich Style Guide](https://github.com/raywenderlich/swift-style-guide)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

7. Architecture / Design Pattern

    - [ ] MVC  
    - [ ] MVP 
    - [ ] MVVM 
    - [ ] VIPER 

8. Architecture Implementation

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

9. SOLID

- Single Responsibility Principle 

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Open Closed Principle

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Liskov substitutability principle

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Interface segregation

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

- Dependency Inversion principle

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

10. Testing

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

11. Force unwrapping !

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

12. Mutable vs immutable property (Right use of `let`  and `var`)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

13. Retain cycles (Capture list)

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

14. Code Documentation

    - [ ] Excellent    
    - [ ] Good    
    - [ ] Adequate
    - [ ] Developing

### CODE REVIEW

File: TheMovieDb/MVP/ViewController.swift [hunk 1]
```diff
@@ -10,7 +10,8 @@ import os.log
 
 final class ViewController: UIViewController {
     
-    var viewModel: MovieListViewModel = .init(facade: MovieFacade())
+    private var presenter: MovieListPresenter?
+    var movieListOption: MoviesOptions = .nowPlaying
```
Comment (Carlos-Barron) [inline TheMovieDb/MVP/ViewController.swift]: This can be private

File: TheMovieDb/MVP/ViewController.swift [hunk 2]
```diff
@@ -61,21 +61,38 @@ final class ViewController: UIViewController {
...
+    }
+   
+    
+    func onUpdateMovies() {
+        DispatchQueue.main.async {
+            self.tableView.reloadData()
+        }
+    }
+}
+
```
Comment (Carlos-Barron) [inline TheMovieDb/MVP/ViewController.swift:98]: Nice

File: TheMovieDbTests/Mocks/MockMovieListView.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+//  MockMovieListView.swift
+//  TheMovieDbTests
+//
+//  Created by Karla Rubiano on 23/11/21.
+//
+
+import Foundation
+@testable import TheMovieDb
+
+class MockMovieListView: MovieListView {
```
Comment (Carlos-Barron) [inline TheMovieDbTests/Mocks/MockMovieListView.swift]: Mocks can be final

**Review by Carlos-Barron — APPROVED** [review]

File: TheMovieDb/MVP/ViewController.swift (hunk 1 above)
Comment (kdrubianoc) [reply TheMovieDb/MVP/ViewController.swift]: Fixed.

File: TheMovieDbTests/Mocks/MockMovieListView.swift (hunk 3 above)
Comment (kdrubianoc) [reply TheMovieDbTests/Mocks/MockMovieListView.swift]: Fixed.

**Pull request merged**